Description: Turn off warning that openfabric is not available
 because it breaks some overlying layers/appications in Debian.
 Also exclude openfabric/openib by default.
 See: #979041
Author: Alastair McKinstry <mckinstry@debian.org>
Last-Updated: 2021-09-02
Forwarded: not-needed

--- a/opal/etc/openmpi-mca-params.conf
+++ b/opal/etc/openmpi-mca-params.conf
@@ -57,3 +57,14 @@
 
 # See "ompi_info --param all all --level 9" for a full listing of Open
 # MPI MCA parameters available and their default values.
+
+# Silence this warning on Debian, as many systems don't have openfabric
+# but the warning breaks higher layers or application
+btl_base_warn_component_unused=0
+# Avoid openib an in case applications use fork: see https://github.com/ofiwg/libfabric/issues/6332
+# If you wish to use openib and know your application is safe, remove the following:
+# Similarly for UCX: https://github.com/open-mpi/ompi/issues/8367
+mtl = ^ofi
+btl = ^uct,openib,ofi
+pml = ^ucx
+osc = ^ucx,pt2pt
\ No newline at end of file
